php-checks/src/main/java/org/sonar/php/checks/security/AuthorizationsCheck.java

```diff
@@ -0,0 +1,198 @@
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.declaration.CallArgumentTree;
+import org.sonar.plugins.php.api.tree.declaration.ClassDeclarationTree;
+import org.sonar.plugins.php.api.tree.declaration.FunctionTree;
+import org.sonar.plugins.php.api.tree.declaration.MethodDeclarationTree;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.tree.expression.MemberAccessTree;
+import org.sonar.plugins.php.api.tree.expression.NameIdentifierTree;
+import org.sonar.plugins.php.api.tree.expression.VariableIdentifierTree;
+import org.sonar.plugins.php.api.tree.statement.ReturnStatementTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+import static java.util.Collections.singletonList;
+import static org.sonar.php.checks.utils.CheckUtils.argument;
+import static org.sonar.php.checks.utils.CheckUtils.getFunctionName;
+import static org.sonar.php.checks.utils.CheckUtils.hasModifier;
+import static org.sonar.php.checks.utils.CheckUtils.nameOf;
+import static org.sonar.php.checks.utils.CheckUtils.trimQuotes;
+import static org.sonar.php.symbols.Symbols.get;
+import static org.sonar.php.tree.TreeUtils.descendants;
+import static org.sonar.php.tree.TreeUtils.findAncestorWithKind;
+import static org.sonar.plugins.php.api.symbols.QualifiedName.qualifiedName;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.BOOLEAN_LITERAL;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.CLASS_MEMBER_ACCESS;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.FUNCTION_CALL;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.FUNCTION_EXPRESSION;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NULL_LITERAL;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NUMERIC_LITERAL;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.REGULAR_STRING_LITERAL;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.VARIABLE_IDENTIFIER;
+
+@Rule(key = "S5808")
+public class AuthorizationsCheck extends PHPVisitorCheck {
+
+  private static final String MESSAGE = "Vote methods should return at least once a negative response";
+
+  private static final String SYMFONY_VOTER_INTERFACE_NAMESPACE = "Symfony\\Component\\Security\\Core\\Authorization\\Voter\\VoterInterface";
+  private static final String SYMFONY_VOTER_NAMESPACE = "Symfony\\Component\\Security\\Core\\Authorization\\Voter\\Voter";
```

Comment:
Maybe init QualifiedName name objects instead of init them on every analyzed call.

Reply:
Good idea.